extension/docs/WebComponentsIntegrationGuide.md

```diff
@@ -1,23 +1,77 @@
-# Integration Guide
-
-The following diagram shows whole checkout flow supported with [Adyen Web Components](https://docs.adyen.com/checkout/components-web).
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents** 
+
+- [Web Components integration guide](#web-components-integration-guide)
+  - [Before you begin](#before-you-begin)
+  - [Step 1: CTP checkout validations](#step-1-ctp-checkout-validations)
+    - [Validate cart state](#validate-cart-state)
+    - [Recalculate cart](#recalculate-cart)
+    - [Validate payment](#validate-payment)
+    - [Validate payment transaction](#validate-payment-transaction)
+  - [Step 2: Get available payment methods](#step-2-get-available-payment-methods)
+  - [Step 3: Add Components to your payments form](#step-3-add-components-to-your-payments-form)
+  - [Step N: Cancel or refund](#step-n-cancel-or-refund)
+  - [Error handling](#error-handling)
+    - [Extension module errors](#extension-module-errors)
+    - [Adyen payment refusals](#adyen-payment-refusals)
+    - [Shopper successfully paid but `redirectUrl` was not reached](#shopper-successfully-paid-but-redirecturl-was-not-reached)
+    - [Shopper tries to pay a different amount than the actual order amount](#shopper-tries-to-pay-a-different-amount-than-the-actual-order-amount)
+  - [Test and go live](#test-and-go-live)
+- [Bad Practices](#bad-practices)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Web Components integration guide
+
+In this integration process, there are different parties working with the extension module. Those: 
+
+- **Shopper** - a person that's using the shop.
+- **Browser** - frontend part of the checkout UI (webshop). 
+- **Merchant Server** - backend part of the checkout.
+- **[Notification module](./../../notification/README.md)** - Notification module is receiving notifications from Adyen payment service provider, processing and storing them on a commercetools payment object.
```

Comment:
No in the diagram there is a notification module flow, I will add an extension module too.

Reply:
I decide to remove these, which is obvious when you check the diagram, now it is forcing someone to think before checking the diagram.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/330/commits/674c51bd03c8493d4d9c09865816810e822f066d

Reply:
OK I see. I would leave the explanation there, because we use these terms in our docs, so it's good to clarify what they means. In this case, just add also `Extension module` explanation. I use it in my steps a lot :)

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/330/commits/8ff0bc7fd35a15e50860604f1503bd0516c080eb